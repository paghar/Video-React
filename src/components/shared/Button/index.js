//@flow
import * as React from 'react';
import { StyledButton, StyledIcon } from './index.style';
type Props = {
  children?: React.Node,
  rtl?: boolean,
  spacing?: string,
  backgroundColor: string,
  width?: string,
  height?: string,
  radius?: string,
  hoverBackgroundColor?: string,
  hoverColor?: string,
  activeBackgroundColor?: string,
  iconColor?: string,
  hoverIconColor?: string,
  iconSize?: string,
  clicked?: () => void,
  icon?: string,
  className?: string,
};
const Button = ({
  icon,
  spacing,
  iconColor,
  children,
  iconSize,
  rtl,
  backgroundColor,
  width,
  height,
  radius,
  hoverBackgroundColor,
  activeBackgroundColor,
  hoverColor,
  clicked,
  className,
}: Props) => {
  return (
    <StyledButton
      rtl={rtl}
      onClick={clicked}
      activeBackgroundColor={activeBackgroundColor}
      backgroundColor={backgroundColor}
      hoverColor={hoverColor}
      width={width}
      height={height}
      radius={radius}
      hoverBackgroundColor={hoverBackgroundColor}
      className={className}
    >
      <StyledIcon
        rtl={rtl}
        color={iconColor}
        fontSize={iconSize}
        spacing={spacing}
        hoverColor={hoverColor}
        iconNumber={icon}
      />
      {children}
    </StyledButton>
  );
};
export default Button;
