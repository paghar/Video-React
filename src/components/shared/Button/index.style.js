//@flow
import styled from 'styled-components';
import Icon from 'components/shared/Icon';
export const StyledIcon = styled(Icon)`
  font-size: ${({ fontSize }) => fontSize};
  display: ${({ fontSize }) => (fontSize ? 'block' : 'none')};
  color: ${({ color }) => color};
  padding-right: ${({ spacing, rtl }) => (rtl ? '0px' : spacing)};
  padding-left: ${({ spacing, rtl }) => (rtl ? spacing : '0px')};
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  user-select: none;
`;

export const StyledButton = styled.button`
  direction: ${({ rtl }) => (rtl ? 'rtl' : 'ltr')};
  outline: none;
  display: flex;
  justify-content: center;
  align-items: center;
  border: none;
  background: none;
  background-color: ${({ backgroundColor }) => backgroundColor};
  width: ${({ width }) => width};
  height: ${({ height }) => height};
  border-radius: ${({ radius }) => radius};
  :hover {
    background-color: ${({ hoverBackgroundColor }) => hoverBackgroundColor};
    span {
      color: ${({ hoverColor }) => hoverColor};
    }
    ${({ hoverIconColor }) =>
      hoverIconColor
        ? `
    ${StyledIcon} {
      color: ${hoverIconColor};
    }
    `
        : ''}
  }
  :active {
    background-color: ${({ activeBackgroundColor }) => activeBackgroundColor};
  }
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  user-select: none;
`;
