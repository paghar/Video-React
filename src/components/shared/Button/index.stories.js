//@flow
import React from 'react';
import { action } from '@storybook/addon-actions';
import Button from '.';
import Typography from 'components/shared/Typography';
import { Display1 } from 'theme/fonts';
import { Primary, Generals } from 'theme/colors';
export default { title: 'Button' };

export const ShareButton = () => (
  <Button
    backgroundColor={Primary.Dim}
    hoverBackgroundColor={Primary.Light}
    radius="5px"
    icon="61920"
    iconColor={Generals.viii}
    spacing="4.8px"
    width="75px"
    iconSize="14px"
    height="34px"
    activeBackgroundColor={Primary.Dark}
    clicked={action('clicked')}
  >
    <Typography font={Display1} color={Generals.viii}>
      Share
    </Typography>
  </Button>
);

export const More = () => (
  <Button
    rtl
    backgroundColor={Generals.i}
    hoverColor={Generals.viii}
    icon="61497"
    iconColor="#c1bbcc"
    spacing="4.8px"
    iconSize="10px"
    width="65px"
    height="29px"
    clicked={action('clicked')}
  ></Button>
);

export const plus3 = () => (
  <Button
    backgroundColor={Generals.vi}
    hoverBackgroundColor={Generals.v}
    activeBackgroundColor={Generals.v}
    radius="50%"
    spacing="0px"
    width="31px"
    height="31px"
    clicked={action('clicked')}
  >
    <Typography font={Display1} color={Generals.viii}>
      +3
    </Typography>
  </Button>
);
export const PlusUser = () => (
  <Button
    rtl
    backgroundColor={Generals.v}
    hoverBackgroundColor={Generals.iv}
    activeBackgroundColor={Generals.iv}
    radius="50%"
    paddingRight="0px"
    icon="61447"
    iconSize="13px"
    iconColor={Generals.vii}
    width="31px"
    height="31px"
    clicked={action('clicked')}
  />
);
export const NameDropDown = () => (
  <Button
    rtl
    backgroundColor={Generals.ii}
    hoverColor={Generals.viii}
    icon="61703"
    iconColor={Generals.vii}
    spacing="4.8px"
    iconSize="10px"
    width="65px"
    height="29px"
    clicked={action('clicked')}
  >
    <Typography font={Display1} color={Generals.vii}>
      Name
    </Typography>
  </Button>
);
export const New = () => (
  <Button
    backgroundColor={Generals.v}
    hoverBackgroundColor={Generals.iv}
    hoverColor={Generals.viii}
    radius="5px"
    iconSize="13px"
    icon="61543"
    spacing="4.8px"
    width="63px"
    height="34px"
    clicked={action('clicked')}
    iconColor={Generals.viii}
  >
    <Typography font={Display1} color={Generals.viii}>
      New
    </Typography>
  </Button>
);
