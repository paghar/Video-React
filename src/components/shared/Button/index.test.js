//@flow
import { shallow, render } from 'enzyme';
import React from 'react';
import Button from '.';
import renderer from 'react-test-renderer';

describe('<Button />', () => {
  it('checks click event', () => {
    const mockCallBack = jest.fn();
    const wrapper = shallow(
      <Button
        backgroundColor="#7b7b9a"
        hoverBackgroundColor="#262640"
        activeBackgroundColor="#262640"
        radius="50%"
        spacing="0px"
        icon="61543"
        iconSize="19px"
        iconColor="#fff"
        width="53px"
        height="53px"
        clicked={mockCallBack}
      />
    );
    wrapper.simulate('click');
    expect(mockCallBack).toHaveBeenCalled();
  });
  it('checks the UI', () => {
    const wrapper = renderer
      .create(
        <Button
          rtl
          backgroundColor="#0d0d1a"
          hoverColor="#fff"
          icon="61703"
          iconColor="#c1bbcc"
          spacing="10px"
          iconSize="20px"
          width="130px"
          height="58px"
        />
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('again checks the UI', () => {
    const wrapper = renderer
      .create(
        <Button
          backgroundColor="#262640"
          hoverBackgroundColor="#212133"
          activeBackgroundColor="#212133"
          radius="50%"
          spacing="0px"
          icon="61543"
          iconSize="20px"
          width="54px"
          height="54px"
        />
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
