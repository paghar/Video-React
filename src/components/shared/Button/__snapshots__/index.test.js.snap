// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> again checks the UI 1`] = `
.c1 {
  font-family: 'FontAwesome' !important;
  font-size: 20px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  font-size: 20px;
  display: block;
  padding-right: 0px;
  padding-left: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  direction: ltr;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background: none;
  background-color: #262640;
  width: 54px;
  height: 54px;
  border-radius: 50%;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:hover {
  background-color: #212133;
}

.c0:active {
  background-color: #212133;
}

<button
  className="c0"
  height="54px"
  radius="50%"
  width="54px"
>
  <span
    className="c1 c2"
    fontSize="20px"
  >
    
  </span>
</button>
`;

exports[`<Button /> checks the UI 1`] = `
.c1 {
  font-family: 'FontAwesome' !important;
  font-size: 20px;
  color: #c1bbcc;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1:hover {
  color: #fff;
}

.c2 {
  font-size: 20px;
  display: block;
  color: #c1bbcc;
  padding-right: 0px;
  padding-left: 10px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  direction: rtl;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background: none;
  background-color: #0d0d1a;
  width: 130px;
  height: 58px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:hover span {
  color: #fff;
}

<button
  className="c0"
  height="58px"
  width="130px"
>
  <span
    className="c1 c2"
    fontSize="20px"
  >
    
  </span>
</button>
`;
